deltacat/compute/compactor/steps/materialize.py

```diff
@@ -230,26 +231,33 @@ def _materialize(compacted_tables: List[pa.Table]) -> MaterializeResult:
             for record_number in record_numbers:
                 record_numbers_length += 1
                 mask_pylist[record_number] = True
-            if (
-                record_numbers_length == src_file_record_count
-                and src_file_partition_locator
-                == round_completion_info.compacted_delta_locator.partition_locator
-            ):
-                logger.debug(
-                    f"Untouched manifest file found, "
-                    f"record numbers length: {record_numbers_length} "
-                    f"same as downloaded table length: {src_file_record_count}"
-                )
-                untouched_src_manifest_entry = manifest.entries[src_file_idx_np.item()]
-                manifest_entry_list_reference.append(untouched_src_manifest_entry)
-                referenced_pyarrow_write_result = PyArrowWriteResult.of(
-                    1,
-                    manifest.meta.source_content_length,
-                    manifest.meta.content_length,
-                    src_file_record_count,
-                )
-                referenced_pyarrow_write_results.append(referenced_pyarrow_write_result)
-            else:
+            can_copy_by_reference = False
+            if round_completion_info and enable_manifest_entry_copy_by_reference:
+                if (
+                    record_numbers_length == src_file_record_count
+                    and src_file_partition_locator
+                    == round_completion_info.compacted_delta_locator.partition_locator
+                ):
+                    logger.debug(
+                        f"Untouched manifest file found, "
+                        f"record numbers length: {record_numbers_length} "
+                        f"same as downloaded table length: {src_file_record_count}"
+                    )
+                    untouched_src_manifest_entry = manifest.entries[
+                        src_file_idx_np.item()
+                    ]
+                    manifest_entry_list_reference.append(untouched_src_manifest_entry)
+                    referenced_pyarrow_write_result = PyArrowWriteResult.of(
+                        1,
+                        untouched_src_manifest_entry.meta.source_content_length,
+                        untouched_src_manifest_entry.meta.content_length,
+                        src_file_record_count,
+                    )
+                    referenced_pyarrow_write_results.append(
+                        referenced_pyarrow_write_result
+                    )
+                    can_copy_by_reference = True
+            if not can_copy_by_reference:
```

Comment:
Can it just be the else part? Why do we need `can_copy_by_reference` var?

Reply:
There are two nested if statement here, 
if condition A:
   if condition B:
One of which being False we will need to run Else part.
`can_copy_by_reference` var set to True only if both if statement is True, so we guarantee any False of two if statement will need to download the manifest.

Reply:
```
if (               enable_manifest_entry_copy_by_reference and round_completion_info and
                    record_numbers_length == src_file_record_count
                    and src_file_partition_locator
                    == round_completion_info.compacted_delta_locator.partition_locator
                ):
else:
      # no skipping
```

Isn't the above and the your code equivalent?

Reply:
This will break when this is no round_completion_info right?

Reply:
@raghumdani, updated